multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/MultiTenantSubscribableMessageSource.java

```diff
@@ -0,0 +1,27 @@
+package org.axonframework.extensions.multitenancy.components;
+
+import java.util.Map;
+
+/**
+ * Interface for multi-tenant message sources that can provide tenant segments.
+ *
+ * @param <T> The type of the tenant segment
+ */
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.SubscribableMessageSource;
+
+/**
+ * Interface for multi-tenant message sources that can provide tenant segments.
+ *
+ * @param <T> The type of the tenant segment, which must extend MessageSource
+ */
```

Comment:
```suggestion
 *
 * @author Stefan Dragisic
 * @since 4.10.0
 * @param <T> The type of the tenant segment, which must extend MessageSource
 */
```